state-reconstruct-fetcher/src/perf_metric.rs

```diff
@@ -0,0 +1,31 @@
+use tokio::time::Duration;
+
+#[derive(Default)]
+pub struct PerfMetric {
+    total: Duration,
+    count: u32,
+}
+
+impl PerfMetric {
+    pub fn add(&mut self, duration: Duration) -> u32 {
+        self.total += duration;
+        self.count += 1;
+        self.count
+    }
+
+    pub fn renew(&mut self) -> String {
```

Comment:
```suggestion
    pub fn reset(&mut self) -> String {
```
This is a bit more applicable I think.

Reply:
OK